### PR TITLE
Patch 12

### PR DESCRIPTION
**Pull request by laszloczol**

### CODE REVIEW

**Comment by valkyrie69** [comment]
Using PR https://github.com/mulesoft/docs-connectors/pull/634

File: modules/ROOT/pages/kafka/kafka-connector-design-center.adoc [hunk 1]
```diff
@@ -68,12 +68,12 @@ Uses all the parameters from <<basiccon>>.
...
-| Key Store Location | The location of the key store file. This is optional and can be used for two-way authentication for the connector.
-| Trust Store Type | The file format of the trust store file.
-| Trust Store Password | The password for the trust store file.
-| Trust Store Location | The location of the trust store file.
+|Key Store Type |(Kerberos SSL) The file format of the key store file. This is optional for a client.
+|Key Store Password |(Kerberos SSL) The store password for the key store file. This is optional for a client and only needed if *Key Store Location* is configured.
+|Key Store Location |(Kerberos SSL) The location of the key store file. This is optional for a client and can be used for two-way authentication for a client.
+|Trust Store Type |(Kerberos SSL) The file format of the trust store file. The default is `JKS`.
+|Trust Store Password |(Kerberos SSL) The password for the trust store file. If a password is not set, access to the truststore is still available, but integrity checking is disabled.
+|Trust Store Location |(Kerberos SSL) Location of the trust store.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-design-center.adoc:76]: We are still mentioning Kerberos which is not supported

File: modules/ROOT/pages/kafka/kafka-connector-design-center.adoc [hunk 2]
```diff
@@ -86,10 +86,10 @@ Uses all the parameters from <<basiccon>>.
 [%header,cols="30s,70a"]
 |===
 | Name | Description
-| Principal | Kerberos principal.
+|Principal |(Kerberos and Kerberos SSL) https://www.google.com/search?q=kerberos+principal[Kerberos principal].
 | Keytab | Path to the keytab file associated with `principal`.
-| Service Name | The Kerberos principal name that the Kafka broker runs as.
-| Additional JAAS Properties | Additional properties as key-value pairs that you need to set in `sasl.jaas.config` and that you usually include in JAAS configuration file.
+|Service name |(Kerberos and Kerberos SSL) The Kerberos principal name that Kafka broker runs as.
+|Additional JAAS Properties |(Kerberos and Kerberos SSL) Additional properties as key-value pairs that you need to set in the `sasl.jaas.config` file and that you can include in Java Authentication and Authorization Service (JAAS) configuration file.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-design-center.adoc:92]: No Kerberos support yet.

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 3]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
 . To create a global element for the connector, set these fields:
 +
-.. Basic:
-** Bootstrap Servers - Comma-separated host-port pairs used for establishing the initial connection to the Kafka cluster. This is the same as the `bootstrap.servers` value you must provide to Kafka clients (producer or consumer).
-** Additional properties - Additional properties as key-value pairs that you need for your connection. Here you can put whatever property Kafka supports.
+* Basic:
+** Bootstrap Servers +
+Comma-separated host-port pairs used for establishing the initial connection to the Kafka cluster. This is the same as the `bootstrap.servers` value you must provide to Kafka clients (producer or consumer).
+** Additional properties +
+Add any additional properties that you need for your connection, and that Kafka supports, as key-value pairs.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:41]: There's no additional properties.

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 4]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Trust Store Password +
+The password for the truststore file.
+** Trust Store Location +
+The location of the truststore file.
 +
 image::kafka/kafka-ssl-studio-config.png[]
 +
-.. Kerberos:
+* For *Kerberos*, configure the Basic fields as well as the following:
 ** All the parameters from the Basic configuration.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:62]: No Kerberos supprt

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 5]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
-** Service Name - The Kerberos principal name that the Kafka broker runs as.
-** Additional JAAS Properties - Additional properties as key-value pairs that you set in `sasl.jaas.config` and that you usually include in JAAS configuration file.
+** Principal +
+The Kerberos principal to which Kerberos can assign tickets.
+** Keytab +
+Path to keytab file associated with the `principal`.
+** Service Name +
+The Kerberos principal name that the Kafka broker runs as.
+** Additional JAAS Properties +
+Additional properties as key-value pairs that you set in `sasl.jaas.config` and that you usually include in JAAS configuration file.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:70]: No Kerberos supprt, no properties

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 6]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+* For *Kerberos*, configure the Basic fields as well as the following:
 ** All the parameters from the Basic configuration.
-** Principal - Kerberos principal
-** Keytab - Path to keytab file associated with the `principal`.
-** Service Name - The Kerberos principal name that the Kafka broker runs as.
-** Additional JAAS Properties - Additional properties as key-value pairs that you set in `sasl.jaas.config` and that you usually include in JAAS configuration file.
+** Principal +
+The Kerberos principal to which Kerberos can assign tickets.
+** Keytab +
+Path to keytab file associated with the `principal`.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:66]: No Kerberos

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 7]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Service Name +
+The Kerberos principal name that the Kafka broker runs as.
+** Additional JAAS Properties +
+Additional properties as key-value pairs that you set in `sasl.jaas.config` and that you usually include in JAAS configuration file.
 +
 image::kafka/kafka-kerberos-studio-config.png[]
 +
-.. Kerberos SSL:
+.. For *Kerberos SSL*, configure the following:
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:74]: No Kerberos support

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 8]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
 image::kafka/kafka-kerberos-ssl-studio-config.png[]
 +
-. Based on the operation what you dragged to the canvas, configure the following fields:
-.. Consumer trigger:
-** Topic - The name of the Kafka topic where to consume messages.
-** Partition offsets (Optional) - list of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
+. Based on the operation that you dragged to the canvas, configure the following fields:
+.. Commit operation:
+** Bean Reference (Optional) +
+The URLs that the consumer can use to try to connect to the Kafka cluster.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:84]: this is not in the commit operation

File: modules/ROOT/pages/kafka/kafka-connector.adoc [hunk 9]
```diff
@@ -35,12 +35,12 @@ You also need access to Kafka to get values for these fields that appear in Stud
...
-|Trust Store Type |(Kerbero SSL) The file format of the trust store file. The default is `JKS`.
-|Trust Store Password |(Kerbero SSL) The password for the trust store file. If a password is not set, access to the truststore is still available, but integrity checking is disabled.
-|Trust Store Location |(Kerbero SSL) Location of the trust store.
+|Key Store Type |(Kerberos SSL) The file format of the key store file. This is optional for a client.
+|Key Store Password |(Kerberos SSL) The store password for the key store file. This is optional for a client and only needed if *Key Store Location* is configured.
+|Key Store Location |(Kerberos SSL) The location of the key store file. This is optional for a client and can be used for two-way authentication for a client.
+|Trust Store Type |(Kerberos SSL) The file format of the trust store file. The default is `JKS`.
+|Trust Store Password |(Kerberos SSL) The password for the trust store file. If a password is not set, access to the truststore is still available, but integrity checking is disabled.
+|Trust Store Location |(Kerberos SSL) Location of the trust store.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector.adoc:43]: No Kerberos support

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 10]
```diff
@@ -93,18 +204,44 @@ The *Message Consumer* operation can be used as an input source in the Kafka con
...
+| Primary Node Only |Whether to execute this source on only the primary node when running in a cluster.
 | Streaming Strategy a| * repeatable-in-memory-stream
 * repeatable-file-store-stream
-* non-repeatable-stream 
+* non-repeatable-stream
 
-Configure to use repeatable streams. 
-| Redelivery Policy a| RedeliveryPolicy -  Defines a policy for processing the redelivery of the same message. 
+Configure to use repeatable streams.
+| Redelivery Policy a| RedeliveryPolicy -  Defines a policy for processing the redelivery of the same message.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:214]: Not available

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 11]
```diff
@@ -93,18 +204,44 @@ The *Message Consumer* operation can be used as an input source in the Kafka con
 | Name | Description
 | Configuration |The name of the configuration to use.
 | Topic |Name of the topic to consume messages from.
-| Partition Offsets a| Array of Offset - List of Offset representing partitions offsets configuration. For each element in the list you have to specify partition index and offset.
-| Primary Node Only |Whether this source should only be executed on the primary node when running in Cluster. 
+| Partition Offsets a| Array of Offset - List of Offset representing partitions offsets configuration. For each element in the list you must specify partition index and offset.
+| Primary Node Only |Whether to execute this source on only the primary node when running in a cluster.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:208]: This is not on the consumer parameters.

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 12]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Bean Reference (Optional) +
+The URLs that the consumer can use to try to connect to the Kafka cluster.
+** Group ID +
+Default group ID for all the Kafka consumers that use this configuration.
+** Offsets (Optional) +
+List of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
+** Topic Subscription Patterns (Optional) +
+The list of subscription regular expressions to subscribe to. Topics are automatically rebalanced according to the number of consumers of the topic.
+** Assignments (Optional) +
+The list of topic-partition pairs to assign. Note that there will be no automatic rebalance of the consumers.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:92]: Exclusive optionals, mention that one of these is required

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 13]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Offsets (Optional) +
+List of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
+** Topic Subscription Patterns (Optional) +
+The list of subscription regular expressions to subscribe to. Topics are automatically rebalanced according to the number of consumers of the topic.
+** Assignments (Optional) + 
+The list of topic-partition pairs to assign. Note that there will be no automatic rebalance of the consumers
++
+image::kafka/kafka-consumer-studio-config.png[]
++
+** Default acknowledgement mode +
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:121]: This is not on the consumer parameters

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 14]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Bean Reference (Optional) +
+The URLs that the consumer can use to try to connect to the Kafka cluster
+** Group ID +
+Default group ID for all the Kafka consumers that use this configuration
+** Offsets (Optional) +
+List of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
+** Topic Subscription Patterns (Optional) +
+The list of subscription regular expressions to subscribe to. Topics are automatically rebalanced according to the number of consumers of the topic.
+** Assignments (Optional) + 
+The list of topic-partition pairs to assign. Note that there will be no automatic rebalance of the consumers
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:117]: This is not on the consumer parameters.

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 15]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Default acknowledgement mode +
+Declares the kind of Acknowledgement mode supported
+** Default poll timeout + 
+The amount of time to block
+** Default poll timeout time unit +
+The time unit for the polling timeout. This combines with pollTimeout to define to total timeout for the polling.
+** Zone ID (Optional) +
+Zone ID is used to convert the provided time stamps into ZonedLocalDateTimes in the results. Default value is the system one.
++
+image::kafka/kafka-producer-studio-config-general.png[]
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:130]: This a reference to the producer, not consumer config

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 16]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Bean Reference (Optional) +
+The URLs that the consumer can use to try to connect to the Kafka cluster
+** Group ID +
+Default group ID for all the Kafka consumers that use this configuration
+** Offsets (Optional) +
+List of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
+** Topic Subscription Patterns (Optional) +
+The list of subscription regular expressions to subscribe to. Topics are automatically rebalanced according to the number of consumers of the topic.
+** Assignments (Optional) +
+The list of topic-partition pairs to assign. Note that there will be no automatic rebalance of the consumers
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:142]: This are not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 17]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
++
+image::kafka/kafka-consumer-studio-config.png[]
++
+** Default acknowledgement mode +
+Declares the type of Acknowledgement mode supported
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:147]: This is not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 18]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Default poll timeout +
+The amount of time unit to block
+** Default poll timeout time unit +
+The time unit for the polling timeout. This combines with pollTimeout to define to total timeout for the polling.
+** Zone ID (Optional) +
+Zone ID is used to convert the provided time stamps into ZonedLocalDateTimes in the results. Default value is the system one.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:153]: This is not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 19]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+image::kafka/kafka-consumer-studio-config.png[]
++
+** Default acknowledgement mode +
+Declares the kind of Acknowledgement mode supported
+** Default poll timeout + 
+The amount of time to block
+** Default poll timeout time unit +
+The time unit for the polling timeout. This combines with pollTimeout to define to total timeout for the polling.
+** Zone ID (Optional) +
+Zone ID is used to convert the provided time stamps into ZonedLocalDateTimes in the results. Default value is the system one.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:128]: This is not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 20]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+image::kafka/kafka-consumer-studio-config.png[]
++
+** Default acknowledgement mode +
+Declares the type of supported Acknowledgement mode
+** Default poll timeout +
+The amount of time unit to block
+** Default poll timeout time unit +
+The time unit for the polling timeout. This combines with pollTimeout to define to total timeout for the polling.
+** Zone ID (Optional) +
+Zone ID is used to convert the provided time stamps into ZonedLocalDateTimes in the results. Default value is the system one.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:103]: This is not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 21]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
-** Topic - The name of the Kafka topic where to consume messages.
-** Partition offsets (Optional) - list of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
+. Based on the operation that you dragged to the canvas, configure the following fields:
+.. Commit operation:
+** Bean Reference (Optional) +
+The URLs that the consumer can use to try to connect to the Kafka cluster.
+** Group ID +
+Default group ID for all the Kafka consumers that use this configuration.
+** Offsets (Optional) +
+List of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:88]: This is not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 22]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+The time unit for the polling timeout. This combines with pollTimeout to define to total timeout for the polling.
+** Zone ID (Optional) +
+Zone ID is used to convert the provided time stamps into ZonedLocalDateTimes in the results. Default value is the system one.
++
+image::kafka/kafka-consumer-studio-config-general.png[]
++
+.. Publish operation:
+** Bean Reference (Optional) +
+The URLs that the consumer can use to try to connect to the Kafka cluster
++
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:160]: This is not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 23]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Bean Reference (Optional) +
+The URLs that the consumer can use to try to connect to the Kafka cluster
+** Group ID +
+Default group ID for all the Kafka consumers that use this configuration
+** Offsets (Optional) +
+List of offsets for configuring partitions. For each element in the list, specify the partition index and offset.
+** Topic Subscription Patterns (Optional) +
+The list of subscription regular expressions to subscribe to. Topics are automatically rebalanced according to the number of consumers of the topic.
+** Assignments (Optional) +
+The list of topic-partition pairs to assign. Note that there will be no automatic rebalance of the consumers
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:178]: This are not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 24]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
 +
 image::kafka/kafka-consumer-studio-config.png[]
 +
-.. Producer operation:
-** Topic - Topic to send the message to.
-** Key - Key belonging to the message that is going to be sent.
-** Message - Message to be sent.
+** Default acknowledgement mode +
+Declares the kind of Acknowledgement mode supported
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:183]: This are not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 25]
```diff
@@ -19,72 +20,182 @@ To configure a connector in Anypoint Studio:
...
+** Default poll timeout +
+The amount of time unit to block
+** Default poll timeout time unit +
+The time unit for the polling timeout. This combines with pollTimeout to define the total timeout for the polling.
+** Zone ID (Optional) +
+Zone ID is used to convert the provided time stamps into ZonedLocalDateTimes in the results. Default value is the system one.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:189]: This are not on the parameters of the operation

File: modules/ROOT/pages/kafka/kafka-connector-studio.adoc [hunk 26]
```diff
@@ -93,18 +204,44 @@ The *Message Consumer* operation can be used as an input source in the Kafka con
...
+Configure to use repeatable streams.
+| Redelivery Policy a| RedeliveryPolicy -  Defines a policy for processing the redelivery of the same message.
 | Reconnection Strategy a| * reconnect
 * reconnect-forever -  A retry strategy in case of connectivity errors.
 |===
 
+
+The *Batch Message Consumer* operation can also be used as an input source in Kafka Connector.
+
+*Message Consumer* fields:
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-studio.adoc:222]: Batch Message Listener

File: modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc [hunk 27]
```diff
@@ -0,0 +1,140 @@
...
+
+* New operations, with new parameters. (see in the: New Operations, Parameters table)
+
+* The Message Listener replaces the Message Consumer Source, it has new parameters are shown in the New Source Table. (see in the: Changed Source table)
+
+* New Source was added (see in the: New Source table)
+
+* Added different Ack modes in the Consumer Configuration AUTO: Mule ACKs the message only if the flow is finished successfully; MANUAL: The user must do the ack manually within the flow; DUPS_OK: Same as the AUTO mode, but the commit is made asynchronously, which can lead to duplicate records; IMMEDIATE: Mule automatically ACKs the message upon reception.
+
+* Added Transactional SASL connection, this connection type is the same as the Producer SASL connection, but with the difference that it’s not a cached connection and is transactional.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc:29]: no kerberos support

File: modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc [hunk 28]
```diff
@@ -0,0 +1,140 @@
...
+
+* Added Transactional plaintext connection, this connection type is the same as the Producer plaintext connection, but with the difference that it’s not a cached connection and is transactional.
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc:31]: There is only only connection type for now.

File: modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc [hunk 29]
```diff
@@ -0,0 +1,140 @@
...
+
+== New Operations, Parameters, and Return Types
+
+[%header%autowidth.spread]
+|===
+|Kafka Operation | Description | Parameters
+
+| Commit
+a| Commits the offsets associated to a message or batch of messages consumed in a Message Listener. |
+** Consumer commit key; ** (The consumer commit key to use to commit the message.)
```
Comment (javiermule) [inline modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc:51]: Add "The commitKey is inserted by the MessageListeners and Consume operation as a parameter of the message"

**Review by javiermule — COMMENTED** [review]
Please review the comments. Thanks.

**Pull request closed without merging**